### PR TITLE
Add method "detach" in utils to address issue #758

### PR DESCRIPTION
**Pull request by faizankshaikh**
## Description

The tutorial for Segmentation assumes that there is a "detach" method in catalyst.utils, but it is not present. Adding the method in utils/torch.py. 

Check the trial run after commit ([gist link](https://colab.research.google.com/gist/faizankshaikh/a89df852ff13af4472008953534e9e8f/segmentation-tutorial.ipynb))

A few things to note before this PR is accepted

- Although the fix is quite straight-forward, there are two changes in catalyst since the issue was reported (abfb121640e6934c99210e9c0f402af5338200c8) and the PR was submitted (f93beff338925cc1bf1b3ff1b32a2c440a4c9427), which in turn demand that the code in the tutorial to be updated. The gist above takes care of both these updates
  - The type(predictions) we get from runner.predict_loader was a numpy ndarray, but changes to generator object
  - runner.prediction_loader doesn't seem to have verbose parameter since the change
- I would sincerely like to adhere to contribution best practices, namely the code styles checks, and writing tests, the current conditions due to the lockdown do not permit me to have the flexibility of coding in my usual workstation. The work I've done for this PR has been tested on colab, as that is the best I can do to review the updates

Let me know if you need further clarifications/updates from my side

## Related Issue

Bug reported here #758 

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [Code of Conduct](https://github.com/catalyst-team/catalyst/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I have read the [Contributing](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md) guide.
- [x] I have checked the code style using `catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [ ] I have written tests for all new methods and classes that I created.
- [ ] I have written the docstring in Google format for all the methods and classes that I used.
- [ ] I have checked the docs using `make check-docs`.
- [x] I have read I need to click 'Login as guest' to see Teamcity build logs.

### CODE REVIEW

**Comment by Scitator** [comment]
dear @faizankshaikh 
By our contribution guidelines, you need to write some docs for new feature :)
Additionally, I suggest you to use python typing to make this function more readable, like
```
def detach(tensor: torch.Tensor) -> np.ndarray:
    return tensor.cpu().detach().numpy()
```

**Comment by faizankshaikh** [comment]
Thanks for the suggestion. I've added docstring & python typing.

Did you get a chance to look at the points I wrote in the previous comment? If yes - do I have to change the tutorial code in jupyter notebook?

Let me know the next steps

**Comment by Scitator** [comment]
Yeah, you are definitely right, we have changed API a bit. So, it would be great if you could also update the tutorial; Somehow, this feature with `predict_loader` have not changed with new release. Sorry about that. And thank you for your perseverance. 👍

**Comment by faizankshaikh** [comment]
Updated the tutorial code. 

I didn't quite get why TeamCity build fails. Can you help me with this?

**Comment by Scitator** [comment]
@faizankshaikh Thank you for the contribution!
TeamCity server has issues with last pytorch version (1.5.0) and its GPUs, so nevermind, we will fix it soon :)
